rj_param_utils/src/ros2_param_provider.cpp

```diff
@@ -0,0 +1,77 @@
+#include <rj_param_utils/ros2_param_provider.h>
+
+namespace params {
+
+ROS2ParamProvider::ROS2ParamProvider(rclcpp::Node* node) {
+    DeclareParameters(node);
+    InitUpdateParamCallbacks(node);
+}
+
+void ROS2ParamProvider::DeclareParameters(rclcpp::Node* node) {
+#define DECLARE_AND_UPDATE_PARAMS(type)                           \
+    for (const auto& [param_name, param] : GetParamMap<type>()) { \
+        rcl_interfaces::msg::ParameterDescriptor descriptor;      \
+        descriptor.description = param->help();                   \
+        const type& val = node->declare_parameter(                \
+            param_name, param->default_value(), descriptor);      \
+        Update(param_name, val);                                  \
+    }
+
+    // Declare the parameters
+    DECLARE_AND_UPDATE_PARAMS(bool)
```

Comment:
These should be untabbed one

Reply:
`clang-format` put these here so I'll leave them here.